src/ConcreteOperations/isdisjoint.jl

```diff
@@ -446,33 +445,37 @@ center, and ``g_i`` are the zonotope's generators.
 For witness production we fall back to a less efficient implementation for
 general sets as the first argument.
 """
-function is_intersection_empty(Z::Zonotope{N},
-                               H::Hyperplane{N},
-                               witness::Bool=false
-                              ) where {N<:Real}
+function is_intersection_empty(Z::Zonotope{N}, H::Hyperplane{N}, witness::Bool=false) where {N<:Real}
     if witness
-        # use less efficient implementation that supports witness production
-        return invoke(is_intersection_empty,
-                      Tuple{LazySet{N}, Hyperplane{N}, Bool},
-                      Z, H, witness)
-    end
-
-    v = H.b - dot(H.a, Z.center)
-    p = ngens(Z)
-    if p == 0
-        abs_sum = zero(N)
+        return _is_intersection_empty(Z, H, Val(true))
     else
-        abs_sum = sum(abs(dot(H.a, Z.generators[:, i])) for i = 1:ngens(Z))
+        return _is_intersection_empty(Z, H, Val(false))
     end
-    return v < -abs_sum || v > abs_sum
 end
 
 # symmetric method
-function is_intersection_empty(H::Hyperplane{N},
-                               Z::Zonotope{N},
-                               witness::Bool=false
-                              ) where {N<:Real}
-    return is_intersection_empty(Z, H, witness)
+is_intersection_empty(H::Hyperplane{N}, Z::Zonotope{N}, witness::Bool=false) where {N<:Real} = is_intersection_empty(Z, H, witness)
+
+function _is_intersection_empty(Z::Zonotope{N}, H::Hyperplane{N}, ::Val{false}) where {N}
+    c, G = Z.center, Z.generators
+    v = H.b - dot(H.a, c)
+
+    n, p = size(G)
+    abs_sum = zero(N)
+    if p > 0
+        @inbounds for i in 1:n
+            aux = zero(N)
```

Comment:
Another idea might be to first check whether `H.a[i]` is not zero. For sparse vectors this might actually pay off. But I am less sure about this suggestion...
```suggestion
            if isapproxzero(H.a[i])
                continue
            end
            aux = zero(N)
```

Reply:
true that if `H.a` is sparse in the inner loop we can iterate over `H.a.nzind`.

Reply:
i'm fine using keeping the current implementation. it can be specialized further on that use case by dispatching on the vector type of the hyperplane.

Reply:
SEV can also be improved, etc